src/library/scala/util/control/ControlThrowable.scala

```diff
@@ -10,30 +10,38 @@
  * additional information regarding copyright ownership.
  */
 
-package scala
-package util.control
+package scala.util.control
 
-/** A marker trait indicating that the `Throwable` it is mixed into is
- *  intended for flow control.
+/** A parent class for exceptions intended for flow control.
  *
- *  Note that `Throwable` subclasses which extend this trait may extend any
- *  other `Throwable` subclass (eg. `RuntimeException`) and are not required
- *  to extend `Throwable` directly.
- *
- *  Instances of `Throwable` subclasses marked in this way should not normally
- *  be caught. Where catch-all behaviour is required `ControlThrowable`
- *  should be propagated, for example:
+ *  Instances of `ControlThrowable` should not normally
+ *  be caught. Where catch-all behaviour is required, `ControlThrowable`
+ *  should be propagated; for example:
  *  {{{
  *  import scala.util.control.ControlThrowable
  *
  *  try {
  *    // Body might throw arbitrarily
  *  } catch {
  *    case c: ControlThrowable => throw c // propagate
- *    case t: Exception        => log(t)  // log and suppress
+ *    case t: Exception        => log(t)  // log and ignore
  *  }
  *  }}}
  *
+ *  Suppression is disabled, because flow control should not suppress
+ *  an exceptional condition.
+ *
+ *  Instances of `ControlThrowable` should not normally have either
+ *  a cause or a writable stack trace.
+ *
+ *  Legacy subclasses may make the stack trace writable by using
+ *  the appropriate constructor. A cause may be set using `initCause`.
+ *
  *  @author Miles Sabin
  */
-trait ControlThrowable extends Throwable with NoStackTrace
+abstract class ControlThrowable private[this] (message: String, writableStackTrace: Boolean) extends RuntimeException(message, null, false, writableStackTrace) {
```

Comment:
Does it break the scala.js use if we have it extend `Throwable` instead? I have a few concerns with it extending `RuntimeException`:

- it's not really an `Exception`
- it will be caught by people catching `Exception`, which though not the recommended thing to do in Scala, is recommended in Java, and *seems* safe

Reply:
No that won't break our use case.

Reply:
I'll put it back to Throwable.